autopilot/agent.go

```diff
@@ -485,211 +486,221 @@ func (a *Agent) controller() {
 		log.Infof("Triggering attachment directive dispatch, "+
 			"total_funds=%v", a.totalBalance)
 
-		// We're to attempt an attachment so we'll obtain the set of
-		// nodes that we currently have channels with so we avoid
-		// duplicate edges.
-		connectedNodes := a.chanState.ConnectedNodes()
-		a.pendingMtx.Lock()
-		nodesToSkip := mergeNodeMaps(a.pendingOpens,
-			a.pendingConns, connectedNodes, a.failedNodes,
-		)
-		a.pendingMtx.Unlock()
-
-		// If we reach this point, then according to our heuristic we
-		// should modify our channel state to tend towards what it
-		// determines to the optimal state. So we'll call Select to get
-		// a fresh batch of attachment directives, passing in the
-		// amount of funds available for us to use.
-		chanCandidates, err := a.cfg.Heuristic.Select(
-			a.cfg.Self, a.cfg.Graph, availableFunds,
-			numChans, nodesToSkip,
-		)
+		err := a.openChans(availableFunds, numChans, totalChans)
 		if err != nil {
-			log.Errorf("Unable to select candidates for "+
-				"attachment: %v", err)
-			continue
+			log.Errorf("Unable to open channels: %v", err)
 		}
+	}
+}
 
-		if len(chanCandidates) == 0 {
-			log.Infof("No eligible candidates to connect to")
-			continue
-		}
+// openChans queries the agent's heuristic for a set of channel candidates, and
```

Comment:
👍 

I think some of the comments and code fragments below can be opened up a bit to take advantage of the new column space.

Reply:
Done!